lib/matplotlib/backends/__init__.py

```diff
@@ -44,53 +87,60 @@ def pylab_setup(name=None):
     show : function
         Show (and possibly block) any unshown figures.
 
+    Raises
+    ------
+    ImportError
+        If a backend cannot be loaded because a different event loop has
+        already started, or if a third-party backend fails to import.
+
     '''
-    # Import the requested backend into a generic module object
     if name is None:
-        # validates, to match all_backends
-        name = matplotlib.get_backend()
-    if name.startswith('module://'):
-        backend_name = name[9:]
-    else:
-        backend_name = 'backend_' + name
-        backend_name = backend_name.lower()  # until we banish mixed case
-        backend_name = 'matplotlib.backends.%s' % backend_name.lower()
-
-    # the last argument is specifies whether to use absolute or relative
-    # imports. 0 means only perform absolute imports.
-    backend_mod = __import__(backend_name, globals(), locals(),
-                             [backend_name], 0)
-
-    # Things we pull in from all backends
-    new_figure_manager = backend_mod.new_figure_manager
-
-    # image backends like pdf, agg or svg do not need to do anything
-    # for "show" or "draw_if_interactive", so if they are not defined
-    # by the backend, just do nothing
-    def do_nothing_show(*args, **kwargs):
-        frame = inspect.currentframe()
-        fname = frame.f_back.f_code.co_filename
-        if fname in ('<stdin>', '<ipython console>'):
-            warnings.warn("""
-Your currently selected backend, '%s' does not support show().
-Please select a GUI backend in your matplotlibrc file ('%s')
-or with matplotlib.use()""" %
-                          (name, matplotlib.matplotlib_fname()))
-
-    def do_nothing(*args, **kwargs):
-        pass
-
-    backend_version = getattr(backend_mod, 'backend_version', 'unknown')
-
-    show = getattr(backend_mod, 'show', do_nothing_show)
-
-    draw_if_interactive = getattr(backend_mod, 'draw_if_interactive',
-                                  do_nothing)
-
-    _log.info('backend %s version %s' % (name, backend_version))
+        name = matplotlib.rcParams["backend"]
+
+    if isinstance(name, list):
```

Comment:
testing if not `str` and and iterable is probably better (so we support anything that ducks as a list here.

Reply:
changed to "not str".